### PR TITLE
Fix HEEx directive injections

### LINKED ISSUE #5191: elixir heex parser fails to parse for loop
### Describe the bug

It seems that this part breaks wrapping for loop directives: https://github.com/nvim-treesitter/nvim-treesitter/blob/master/queries/heex/injections.scm#L1-L5

playground output in neovim:
image

playground output outside neovim:
image

### To Reproduce

1. Install heex grammar
2. make a file foo.html.heex containing:

```heex
<ul>
  <%= for item <- @image do %>
    <li></li>
  <% end %>
</ul>
```

### Expected behavior

no error in parsing

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.20.8 (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v18.16.1 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: Apple clang version 14.0.3 (clang-1403.0.22.14.1)
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "arm64",
  release = "22.5.0",
  sysname = "Darwin",
  version = "Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000"
} ~

Parser/Features         H L F I J
  - awk                 ✓ . . . ✓
  - bash                ✓ ✓ ✓ . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - dockerfile          ✓ . . . ✓
  - eex                 ✓ . . . ✓
  - elixir              ✓ ✓ ✓ ✓ ✓
  - erlang              ✓ . ✓ . .
  - git_config          ✓ . . . .
  - gitignore           ✓ . . . .
  - gleam               ✓ ✓ ✓ ✓ ✓
  - go                  ✓ ✓ ✓ ✓ ✓
  - gomod               ✓ . . . ✓
  - gosum               ✓ . . . .
  - gowork              ✓ . . . ✓
  - heex                ✓ ✓ ✓ ✓ ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - jq                  ✓ . . . ✓
  - json                ✓ ✓ ✓ ✓ .
  - json5               ✓ . . . ✓
  - jsonc               ✓ ✓ ✓ ✓ ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - luadoc              ✓ . . . .
  - luap                ✓ . . . .
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - po                  ✓ . ✓ . ✓
  - proto               ✓ . ✓ . .
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - regex               ✓ . . . .
  - ron                 ✓ ✓ ✓ ✓ ✓
  - ruby                ✓ ✓ ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - scala               ✓ ✓ ✓ . ✓
  - scss                ✓ . ✓ ✓ .
  - sql                 ✓ . . ✓ ✓
  - surface             ✓ . ✓ ✓ ✓
  - toml                ✓ ✓ ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓
  - vue                 ✓ . ✓ ✓ ✓
  - yaml                ✓ ✓ ✓ ✓ ✓
  - zig                 ✓ . ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~
```

### Output of `nvim --version`

```text
NVIM v0.9.1
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.9.1/share/nvim"

Run :checkhealth for more info
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dvic**
Fixes #5191

copied from https://github.com/phoenixframework/tree-sitter-heex/tree/main/queries

### CODE REVIEW

**Comment by dvic** [comment]
Not sure why the checks are failing, this is the error: 

```
./scripts/check-queries.lua:96: (x) Invalid capture @injection.content in injections for heex.
```

**Comment by clason** [comment]
You're using upstream injection format, which is not supported by Neovim <0.9 (which we still support). Check the other injection queries for the legacy format.

**Comment by clason** [comment]
(Or wait until we drop support for 0.8.x.)

**Comment by dvic** [comment]
Clear, so if I understand correctly, the 'include-children' part also won't work? 

It's not mentioned in https://github.com/neovim/neovim/blob/v0.8.3/runtime/doc/treesitter.txt but it is in https://github.com/neovim/neovim/blob/v0.9.0/runtime/doc/treesitter.txt

**Comment by dvic** [comment]
forgot to update the highlights as well: `@module` -> `@type` I guess.

File: queries/heex/injections.scm [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+;     <% end %>
+((directive (partial_expression_value) @elixir @combined))
 
-; HEEx Elixir expressions are always within a tag or component
-(expression (expression_value) @elixir)
+; Regular expression_values do not need to be combined
+((directive (expression_value) @elixir))
 
-; HEEx comments
-(comment) @comment
```
Comment (clason) [inline queries/heex/injections.scm:11]: why did you remove the comment injection?

File: queries/heex/highlights.scm [hunk 2]
```diff
@@ -29,9 +29,6 @@
 ; HEEx comments are highlighted as such
 (comment) @comment
 
-; HEEx text content is treated as markup
-(text) @text
```
Comment (clason) [inline queries/heex/highlights.scm:33]: why did you remove this query?

**Review by clason — CHANGES_REQUESTED** [review]
There's a bunch of unrelated changes here that seem to be just overwriting our (deliberately different) queries with upstream ones.

Please only make the change(s) necessary to fix the mentioned issue, keeping in mind the different capture names and query order precedence in this project.

**Comment by dvic** [comment]
> There's a bunch of unrelated changes here that seem to be just overwriting our (deliberately different) queries with upstream ones.
> 
> Please only make the change(s) necessary to fix the mentioned issue, keeping in mind the different capture names and query order precedence in this project.

Ah okay, I was not aware that these differences were deliberate so I just brought everything in sync with keeping in mind the different capture names.

I'll update this PR to do the bare minimum that fixes the issue.

**Comment by clason** [comment]
If there are any other changes that look positive, feel free to make a separate PR to include them! (And vice-versa, upstream).

File: queries/heex/injections.scm (hunk 1 above)
Comment (dvic) [reply queries/heex/injections.scm:11]: @clason is it needed in the injections? it's already in the highlights? i'm really a newbie with regards to treesitter syntax, i'll have to read up on that but it seems it's redundant here.

Comment (clason) [reply queries/heex/injections.scm:11]: Yes, it's wanted, to be able to apply `TODO` etc. highlighting in comments.

(If in doubt, don't touch it ;))

**Comment by dvic** [comment]
@clason I've rebased against the main branch (I assume this is the path forward, with the new syntax) and made only the necessary injection changes.

File: runtime/queries/heex/injections.scm [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
-  (expression_value) @injection.content 
-  (#set! injection.language "elixir"))
+; Regular expression_values do not need to be combined
+((directive (expression_value) @injection.content)
+ (#set! injection.language "elixir"))
+
+; expressions live within HTML tags, and do not need to be combined
+;     <link href={ Routes.static_path(..) } />
+((expression (expression_value) @injection.content)
+ (#set! injection.language "elixir"))
```
Comment (amaanq) [inline runtime/queries/heex/injections.scm:20]: Seems like the last one was unchanged?

Comment (dvic) [reply runtime/queries/heex/injections.scm:20]: Looks like it, yes, but should I then also change the formatting of the two directive parts above? This is how it's formatted upstream https://github.com/phoenixframework/tree-sitter-heex/blob/main/queries/injections.scm

Comment (clason) [reply runtime/queries/heex/injections.scm:20]: Don't make unrelated formatting changes, please; these make it harder to review. (And long-term, we want to run a formatter over all of them anyway.)

**Comment by clason** [comment]
> @clason I've rebased against the main branch (I assume this is the path forward, with the new syntax) and made only the necessary injection changes.

The `main` branch is still quite WIP, so this means pretty much nobody will see those changes. Better to make the PR against `master` with the old syntax. (I'll adapt them when rebasing.)

**Comment by clason** [comment]
Or we pull the plug now and bump the supported Neovim version so we can switch to new-style injections across the board; that would make things easier.

**Comment by dvic** [comment]
> > @clason I've rebased against the main branch (I assume this is the path forward, with the new syntax) and made only the necessary injection changes.
> 
> The `main` branch is still quite WIP, so this means pretty much nobody will see those changes. Better to make the PR against `master` with the old syntax. (I'll adapt them when rebasing.)

Done!

> Or we pull the plug now and bump the supported Neovim version so we can switch to new-style injections across the board; that would make things easier.

Yeah I'd have no objections to that :)

File: queries/heex/injections.scm [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
+;       <p>hello, tree-sitter!</p>
+;     <% end %>
+(directive (partial_expression_value) @elixir @combined)  
 
-; HEEx Elixir expressions are always within a tag or component
+; Regular expression_values do not need to be combined
+(directive (expression_value) @elixir)
+
+; expressions live within HTML tags, and do not need to be combined
+;     <link href={ Routes.static_path(..) } />
```
Comment (dvic) [inline queries/heex/injections.scm:14]: For easy reviewing, a reference of the upstream version with updated syntax:

https://github.com/phoenixframework/tree-sitter-heex/blob/main/queries/injections.scm

image

Comment (clason) [reply queries/heex/injections.scm:14]: Could you please just paste this as a codeblock? Screenshots are hard to work with (e.g., for later copy&pasting).

Comment (dvic) [reply queries/heex/injections.scm:14]: done

Comment (clason) [reply queries/heex/injections.scm:14]: And now _we_ switched to the upstream format, so you can rebase on `master` and use the upstream version :)

Comment (dvic) [reply queries/heex/injections.scm]: Done :)

**Review by amaanq — APPROVED** [review]

**Review by clason — APPROVED** [review]
